src/main/java/org/spongepowered/api/text/TextBuilder.java

```diff
@@ -0,0 +1,1124 @@
+                    this.clickAction,
+                    this.hoverAction,
+                    this.shiftClickAction,
+                    this.content);
+        }
+
+        @Override
+        public boolean equals(@Nullable Object o) {
+            if (this == o) {
+                return true;
+            }
+            if (!(o instanceof Literal) || !super.equals(o)) {
+                return false;
+            }
+
+            Literal that = (Literal) o;
+            return Objects.equal(this.content, that.content);
+
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(super.hashCode(), this.content);
+        }
+
+        @Override
+        public String toString() {
+            return MoreObjects.toStringHelper(this)
+                    .addValue(super.toString())
+                    .add("content", this.content)
+                    .toString();
+        }
+
+        @Override
+        public Literal color(TextColor color) {
+            return (Literal) super.color(color);
+        }
+
+        @Override
+        public Literal style(TextStyle... styles) {
+            return (Literal) super.style(styles);
+        }
+
+        @Override
+        public Literal onClick(@Nullable ClickAction<?> clickAction) {
+            return (Literal) super.onClick(clickAction);
+        }
+
+        @Override
+        public Literal onHover(@Nullable HoverAction<?> hoverAction) {
+            return (Literal) super.onHover(hoverAction);
+        }
+
+        @Override
+        public Literal onShiftClick(@Nullable ShiftClickAction<?> shiftClickAction) {
+            return (Literal) super.onShiftClick(shiftClickAction);
+        }
+
+        @Override
+        public Literal append(Text... children) {
+            return (Literal) super.append(children);
+        }
+
+        @Override
+        public Literal append(Iterable<? extends Text> children) {
+            return (Literal) super.append(children);
+        }
+
+        @Override
+        public Literal insert(int pos, Text... children) {
+            return (Literal) super.insert(pos, children);
+        }
+
+        @Override
+        public Literal insert(int pos, Iterable<? extends Text> children) {
+            return (Literal) super.insert(pos, children);
+        }
+
+        @Override
+        public Literal remove(Text... children) {
+            return (Literal) super.remove(children);
+        }
+
+        @Override
+        public Literal remove(Iterable<? extends Text> children) {
+            return (Literal) super.remove(children);
+        }
+
+        @Override
+        public Literal removeAll() {
+            return (Literal) super.removeAll();
+        }
+
+    }
+
+    /**
+     * Represents a {@link TextBuilder} creating immutable
+     * {@link Text.Translatable} instances.
+     *
+     * @see Text.Translatable
+     */
+    public static class Translatable extends TextBuilder {
```

Comment:
It's available directly through the translation setters, because there is no point in adding them separately.